ts/components/messages/paginated/MessageList/index.tsx

```diff
@@ -222,25 +223,58 @@ const MessageList = ({
     }
   };
 
+  // This component has two different spinners: one on top and one
+  // on bottom. The former is the standard refresh control while the
+  // latter is set as a footer to the list, but only when it has to
+  // be shown.
+  // There are three data events that require one of them to be shown:
+  // the loading of the next message page (older messages), the
+  // loading of the previos message page (newer messager) or a global
+  // reloading of all messages.
+  // While next messages are only loaded when the end of the list has
+  // been reached by scrolling (and there are more messages to load),
+  // the other two loadings are triggered by using the refresh
+  // gesture or when the refresh interval is hit or when a push
+  // notification is received and selected with the app in foreground
+  // while showing this list or, lastly, at the first showing of this
+  // component.
+  // In order not to show both spinners, the footer one is shown only
+  // if loading the next message page (older messages) or if there
+  // are no messages in the list and the loading was triggered
+  // automatically (i.e., not from an user swipe-down-to-refresh).
+  // The top spinner, on the other hand, is shown only if the user
+  // request a loading by swiping-down-to-refresh or, when the
+  // loading was system initiated, if there were already messages in
+  // the list. Note that the top spinnmer is never shown when
+  // loading next page messages (since they are loaded when the list
+  // end has been reached. In that case, the footer spinner is shown)
+  const isLoadingPreviousOrAll = isLoadingPrevious || isReloadingAll;
+  const shouldShowFooterLoader =
+    isLoadingMore ||
+    (!hasMessages && !isRefreshFromUser && isLoadingPreviousOrAll);
+  const shouldShowTopRefreshControl =
+    isRefreshFromUser || (hasMessages && isLoadingPreviousOrAll);
+  const isLoadingOrRefreshingMessageList =
+    isLoadingMore || isLoadingPreviousOrAll;
+
```

Comment:
If we move all these conditions at the top of the component should we use `isLoadingPreviousOrAll` also on the first `useEffect` at line  `167`?

Reply:
`const isLoadingPreviousOrAll = isLoadingPrevious || isReloadingAll;` has been moved above as suggested

Reply:
I've also updated the cases-table to properly address the correct expected results for row 5 and row 6 cases. As you said, <case 5, bottom spinner> and <case 6, top spinner> shall never happen.

Reply:
<case 4, top spinner> seems to be happening only on the iOS simulator. I tested it on an iPhone 6s, iOS 15.7 and the spinner appears correctly.

My suggestion is to take note of this behaviour and check it again on the testflight build.

Reply:
Perfect! So I put here below the updated table with all use cases:

| &nbsp; | Top Spinner | Bottom Spinner |
| - | - | - |
| 01. ✅ Reload all messages on first rendering | never | <video src="https://user-images.githubusercontent.com/16268789/208061471-a5c6a923-0651-4fa7-8d2a-36e9a2e4bb8d.mp4" /> |
| 02.  ✅  Reload all messages on pull to refresh | <video src="https://user-images.githubusercontent.com/16268789/208062070-3dfc61af-317d-4dea-bec8-36891057bad2.mp4" /> | never |
| 03. ✅  Reload next page on the end of the current page list (more pages) | never | <video src="https://user-images.githubusercontent.com/16268789/208063186-f0d18ca3-2d81-4219-b0b0-89cec1450ea6.mp4" /> |
| 04. ✅ Reload previous page on refresh interval (switch tabs) | <video src="https://user-images.githubusercontent.com/16268789/208064263-be826501-5bb2-4cdc-8592-9a93223b3305.mp4" /> | <video src="https://user-images.githubusercontent.com/16268789/208064334-32cc2775-0b4a-4086-82a0-ad94ec803fc8.mp4" /> |
| 04. iOS (⚠️) Reload previous page on refresh interval (coming from another section) |  ❌ (iOS) <video src="https://user-images.githubusercontent.com/16268789/208065643-b9d5be7f-e607-4dea-9c2a-ddb80ecd9810.mp4" /> | ✅ <video src="https://user-images.githubusercontent.com/16268789/208064974-d53acbaa-0dd2-4949-b0e0-f19558ea7fe8.mp4" /> |
| 04. Android (✅ ) Reload previous page on refresh interval (coming from another section) |  <video src="https://user-images.githubusercontent.com/16268789/208066667-e1a3a1e1-807e-4d57-b38a-74d492ab7646.mov" /> | &nbsp; |
| 05. ✅ Reload previous page on foreground push | <video src="https://user-images.githubusercontent.com/16268789/208397201-741e00dd-e8ce-4979-a553-c0caf8c668cd.mp4" /> | Never | 
| 06. ✅  Reload all messages on foreground push | Never | <video src="https://user-images.githubusercontent.com/16268789/208067463-8efc92a1-1393-48c3-854f-2a94f3431d4f.mp4" /> 

As for the spinner not showing behavior on use case number 4, it could also depend on the iOS version, but I agree with you to take note of it and check it again on the TestFlight build.